app/filters/contact_method_filter.rb

```diff
@@ -0,0 +1,13 @@
+class ContactMethodFilter < BasicFilter
+  def self.options
+    {
+      name: 'Contact Methods',
+      filters: FilterOptionsBlueprint.render_as_hash(ContactMethod.enabled.distinct(:name))
+    }
+  end
+
+  def filter(scope)
+    return super unless parameters
+    scope.joins(:person).where(people: { preferred_contact_method: parameters.keys })
```

Comment:
I'm noticing all the filters use `parameters.keys`. Could we get the keys in `BrowseFilter` or `BaseFilter.new` and make just the keys available to individual filters?

Relatedly, the shape of `parameters` is not clear from looking at the code. Could be solved by looking at specs, or maybe a comment describing how params will come in?

Reply:
> the shape of parameters is not clear from looking at the code

This is really the key point here. Thanks for surfacing this, because this point is helping me find the most confusing points.

I started writing some tests around this and got stuck because it was *hard*. I think I've introduced a small, modest mess in `BrowseFilter#contributions` and cleaning up the parameters structure here is dependent on cleaning that up, too.

There's a part of me that wants to get this PR out first and then focus on cleaning up that problem. I want to do that because this PR has taken a very long time and not directly for any code-specific reasons. This PR seems so big to me already, and I'd really like to focus on the refactor I'll need here as a separate thing from the other changes in this PR.

I'm open to suggestions. Let me know what you think